### PR TITLE
move action scheduler to external via composer

### PR DESCRIPTION
**Pull request by rrennick**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR removes the library copy of Action Scheduler and adds it as an external through composer. The version of AS used in this PR is the same version as was in the `libraries` folder (2.2.5). This will ensure that any issues found in testing are related to the change in inclusion. AS can be updated to 3.0.X in a follow up PR.

### How to test the changes in this Pull Request:

1. Run `composer install`
2. Add a webhook to `order.create`
3. Create an order
4. Verify that the pending webhook action is created in Tools -> Scheduled Actions
5. Run `wp action-scheduler run`
6. Verify the action was completed and the webhook processed to the destination.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -22,6 +23,7 @@
       "php": "7.1"
     },
     "preferred-install": {
+        "woocommerce/action-scheduler": "source",
```
Comment (kloon) [inline composer.json]: Should we perhaps also use dist here?

**Review by kloon — DISMISSED** [review]
@rrennick a quick look at the Action Scheduler package it seems it is not using the vendor autoloader file to load it, I might have missed it if it does use it somewhere else but if we want to run this as a package we need to ensure we use the autoloader file. See https://github.com/woocommerce/woocommerce-example-package/blob/master/woocommerce-example-package.php as an example.

File: composer.json (hunk 1 above)
Comment (rrennick) [reply composer.json]: @kloon Thanks :) Yes, I'll update that to `dist`. I'm doing a similar PR https://github.com/woocommerce/woocommerce-admin/pull/3557. I'll leave this one on needs feedback for now.

**Review by rrennick — COMMENTED** [review]
>  it seems it is not using the vendor autoloader file to load it, I might have missed it if it does use it somewhere else but if we want to run this as a package we need to ensure we use the autoloader file.

The main AS file is procedural functions that register the version AS which allows implementing AS in multiple active plugins and having the highest registered version used. If AS is switched to be autoloaded by the parent plugin then the AS version that will be active is the one in the first plugin alphabetically (due to the way autoloading works).

**Comment by peterfabian** [comment]
Note: pls check text domain and translations.

File: composer.json (hunk 1 above)
Comment (rrennick) [reply composer.json]: Done in `9b99f3b`.

**Comment by rrennick** [comment]
This PR won't pass the E2E tests until #25536 is merged. In a later PR, we can bump the AS version to 3.0.X.

**Comment by peterfabian** [comment]
I tested this a couple of times and it works fine, except for the translation part (which we need to fix for both PHP and JS files). So, I think this one is almost good to go.

**Comment by peterfabian** [comment]
Thanks @rrennick , I think we'll solve the textdomain issues in another PR, as we need to update some other issues in how i18n is going to be handled.

**Review by peterfabian — APPROVED** [review]
Thanks @rrennick , I think we'll solve the textdomain issues in another PR, as we need to update some other issues in how i18n is going to be handled.

**Pull request merged**